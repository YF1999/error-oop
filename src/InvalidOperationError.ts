import {
    ErrorOptions,
    InvalidOperationErrorMessageProps,
    InvalidOperationErrorProps,
} from './CommonTypes';
import { AbstractError } from './NativeErrors';

export abstract class AbstractInvalidOperationError extends AbstractError {
    public constructor(
        props: InvalidOperationErrorProps,
        options: ErrorOptions<InvalidOperationErrorMessageProps>,
    ) {
        super(props, options);
    }
}

/**
 * Applicable when a function call or a method call is invalid for the object's current state.
 */
export class InvalidOperationError extends AbstractInvalidOperationError {
    public constructor();
    /**
     * @param message The error message that explains the reason for this error.
     */
    public constructor(message: string);
    /**
     * @param message The error message that explains the reason for this error.
     * @param innerError The error that is the cause of the current error. Stack trace will be
     * appended.
     */
    public constructor(message: string, innerError: Error);

    public constructor(message: string = '', innerError?: Error) {
        super({ message, innerError }, {});
    }
}
