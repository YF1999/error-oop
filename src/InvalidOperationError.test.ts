import { InvalidOperationError } from './InvalidOperationError';

standardTestError(InvalidOperationError, 'InvalidOperationError', 'error');
