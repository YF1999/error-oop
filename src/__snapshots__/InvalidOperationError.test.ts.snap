// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test \`InvalidOperationError\` own keys 1`] = `
Array [
  "stack",
]
`;

exports[`Test \`InvalidOperationError\` own keys 2`] = `
Array [
  "stack",
  "message",
]
`;

exports[`Test \`InvalidOperationError\` own keys 3`] = `
Array [
  "stack",
  "message",
]
`;

exports[`Test \`InvalidOperationError\` own keys 4`] = `
Array [
  "stack",
  "message",
]
`;
